### PR TITLE
Some quality of life improvements

### PR DESCRIPTION
**Pull request by Kurupapuru**
Added `BlocksClient` to `NotionClient`, not sure why it wasn't there

Added `GetTitle` to `Page` as extension
Added `GetAllPlainText` to `TitlePropertyValue` as extension
Added `QueryToEndAsync` to `IDatabaseClient` as extension

Added method `CreateCopy` to `DatabasesQueryParameters` for `QueryToEndAsync` method, not sure if that is right way

I understand that something can be questionable and I'm new to contributing so feel free to point at my mistakes

### CODE REVIEW

File: Src/Notion.Client/Extensions/QualityOfLifeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Notion.Client.Extensions
+{
+    public static class QualityOfLifeExtensions
```
Comment (KoditkarVedant) [inline Src/Notion.Client/Extensions/QualityOfLifeExtensions.cs:9]: @Kurupapuru I'm sure there could be many more helpful extensions we may want to add. So I think it is better if we create a separate library (NuGet) to maintain them. So the user can inject them as they require. Let me know your views?

File: Src/Notion.Client/Api/Databases/RequestParams/DatabasesQueryParameters.cs [hunk 2]
```diff
@@ -8,5 +8,16 @@ public class DatabasesQueryParameters : IDatabaseQueryBodyParameters
         public List<Sort> Sorts { get; set; }
         public string StartCursor { get; set; }
         public string PageSize { get; set; }
+
+        public DatabasesQueryParameters CreateCopy()
```
Comment (KoditkarVedant) [inline Src/Notion.Client/Api/Databases/RequestParams/DatabasesQueryParameters.cs:12]: If this is only required by extensions then you can move it there. Apart from that we may need deep copy of an object. [ref](https://docs.microsoft.com/en-us/dotnet/api/system.object.memberwiseclone?view=net-5.0)

File: Src/Notion.Client/NotionClient.cs [hunk 3]
```diff
@@ -6,6 +6,7 @@ public interface INotionClient
         IDatabasesClient Databases { get; }
         IPagesClient Pages { get; }
         ISearchClient Search { get; }
+        IBlocksClient Blocks { get; }
```
Comment (KoditkarVedant) [inline Src/Notion.Client/NotionClient.cs:9]: @Kurupapuru if you could create a separate PR for this fix would be great so I can merge that and create a release.

Comment (KoditkarVedant) [reply Src/Notion.Client/NotionClient.cs:9]: Thank you for catching this.

**Review by KoditkarVedant — CHANGES_REQUESTED** [review]
Appreciate the PR - there are few review comments which you might wanna have look at them.

File: Src/Notion.Client/Extensions/QualityOfLifeExtensions.cs (hunk 1 above)
Comment (Kurupapuru) [reply Src/Notion.Client/Extensions/QualityOfLifeExtensions.cs:9]: @KoditkarVedant sorry for such late response, expected to get notification from github
I think that is a good option. I don't have any experience with creating projects for publishing nuget packages, but if you will create repository with needed structure I can make pull request there.

File: Src/Notion.Client/NotionClient.cs (hunk 3 above)
Comment (Kurupapuru) [reply Src/Notion.Client/NotionClient.cs:9]: Saw that this has already been fixed, again sorry for late response
#35

File: Src/Notion.Client/Api/Databases/RequestParams/DatabasesQueryParameters.cs (hunk 2 above)
Comment (Kurupapuru) [reply Src/Notion.Client/Api/Databases/RequestParams/DatabasesQueryParameters.cs:12]: Yep, just for extension

File: Src/Notion.Client/Extensions/QualityOfLifeExtensions.cs (hunk 1 above)
Comment (KoditkarVedant) [reply Src/Notion.Client/Extensions/QualityOfLifeExtensions.cs:9]: @Kurupapuru I've created a repository so you can add these extensions there.

**Repo:**  https://github.com/notion-dotnet/notion-sdk-extensions

**Comment by KoditkarVedant** [comment]
Hi @Kurupapuru, I've created a repository so you can add these extensions there. I am closing this PR.

**Repo:**  https://github.com/notion-dotnet/notion-sdk-extensions

**Pull request closed without merging**